src/NUnitFramework/framework/Constraints/CollectionOrderedConstraint.cs

```diff
@@ -103,7 +109,55 @@ public CollectionOrderedConstraint Using<T>(Comparison<T> comparer)
         /// </summary>
         public CollectionOrderedConstraint By(string propertyName)
         {
-            this.propertyName = propertyName;
+            if (this.properties != null && this.properties.Count != 0)
+                throw new InvalidOperationException("Cannot use By on existing collection. Use ThenBy to add additional items.");
```

Comment:
Why not? Seems like we can figure out what is intended.

Even if we want to forbid it, the message needs to be more user oriented. The user knows nothing about the internals of our implementation, so it would have to say something like "Cannot use By more than once" 

However, I see no reason to forbid it, since it's clear what the user wanted to do. Seems as if By and ThenBy can actually work just the same, with the two spellings simply serving to make things more readable.